internal/cloud/azure/application_permission.go

```diff
@@ -20,132 +23,117 @@ type ApplicationPermissionConfig struct {
 	Scope              string
 }
 
-// type RoleAssignmentsClient interface {
-// 	Create(ctx context.Context, roleAssignment msgraph.UnifiedRoleAssignment) (*msgraph.UnifiedRoleAssignment, int, error)
-// 	Get(ctx context.Context, id string, query odata.Query) (*msgraph.UnifiedRoleAssignment, int, error)
-// 	Delete(ctx context.Context, id string) (int, error)
-// }
-
-// func (c *AzureConfig) NewRoleAssignmentsClient(ctx context.Context) (RoleAssignmentsClient, error) {
-// 	authorizer, authorizerErr := c.authConfig.NewAuthorizer(ctx, c.authConfig.Environment.MsGraph)
-// 	if authorizerErr != nil {
-// 		return nil, authorizerErr
-// 	}
-// 	raClient := msgraph.NewRoleAssignmentsClient(c.Provider.TenantID.Value)
-// 	raClient.BaseClient.Authorizer = authorizer
-// 	return raClient, nil
-// }
-
-// type RoleDefinitionsClient interface {
-// 	Get(ctx context.Context, id string, query odata.Query) (*msgraph.UnifiedRoleDefinition, int, error)
-// 	List(ctx context.Context, query odata.Query) (*[]msgraph.UnifiedRoleDefinition, int, error)
-// }
-
-// func (c *AzureConfig) NewRoleDefinitionsClient(ctx context.Context) (RoleDefinitionsClient, error) {
-// 	authorizer, authorizerErr := c.authConfig.NewAuthorizer(ctx, c.authConfig.Environment.ResourceManager)
-// 	if authorizerErr != nil {
-// 		return nil, authorizerErr
-// 	}
-// 	rdClient := msgraph.NewRoleDefinitionsClient(c.Provider.TenantID.Value)
-// 	rdClient.BaseClient.Authorizer = authorizer
-// 	return rdClient, nil
-// }
+func getAzureResourceManagerAuthorizer(ctx context.Context, c *AzureConfig) (autorest.Authorizer, error) {
```

Comment:
This is a really dumb headache. Authorization in the new client library is a lot easier, but this is what we are stuck with on the old client.  Don't waste brain cells reviewing this, its basically copied from [here](https://github.com/hashicorp/terraform-provider-azurerm/blob/accdeebe6072390c531b88f8e322b80acceecfb8/internal/clients/builder.go#L40-L95).